fs/binfs/fs_binfs.c

```diff
@@ -469,21 +469,35 @@ static int binfs_stat(struct inode *mountpt,
                       const char *relpath, struct stat *buf)
 {
   finfo("Entry\n");
+  int index;
+#ifdef CONFIG_SCHED_USER_IDENTITY
+  int mode;
+#endif
 
   /* The requested directory must be the volume-relative "root" directory */
 
   if (relpath && relpath[0] != '\0')
     {
       /* Check if there is a file with this name. */
 
-      if (builtin_isavail(relpath) < 0)
+      index = builtin_isavail(relpath);
+      if (index < 0)
         {
           return -ENOENT;
         }
 
       /* It's a execute-only file name */
 
       buf->st_mode = S_IFREG | S_IXOTH | S_IXGRP | S_IXUSR;
+
+#ifdef CONFIG_SCHED_USER_IDENTITY
+      buf->st_uid = builtin_getuid(index);
+      buf->st_gid = builtin_getgid(index);
+
+      mode = builtin_getmode(index);
+      buf->st_mode |= (mode & S_ISUID) ? S_ISUID : 0;
```

Comment:
```suggestion
      buf->st_mode = builtin_getmode(index);
```
and remove mode variable